File: Task/Two-identical-strings/Perl/two-identical-strings.pl
#!/usr/bin/perl

use strict; # https://rosettacode.org/wiki/Two_identical_strings
use warnings;

while( 1 )
  {
  my $binary = ( sprintf "%b", ++$- ) x 2;
  (my $decimal = oct "b$binary") >= 1000 and last;
  printf "%4d  %s\n", $decimal, $binary;
  }

File: Task/Two-identical-strings/Ruby/two-identical-strings.rb
for i in 0 .. 1000
    bin = i.to_s(2)
    if bin.length % 2 == 0 then
        half = bin.length / 2
        if bin[0..half-1] == bin[half..] then
            print "%3d: %10s\n" % [i, bin]
        end
    end
end
